app/Controllers/Http/UserController.js

```diff
@@ -77,6 +125,26 @@ class UserController {
 		}
 	}
 
+	/**
+	 * Render a specific edit user page depending on the user Id.
+	 *
+	 * @param {Object} Context The context object.
+	 */
+	async edit ({ params, view, auth }) {
+		// Retrieves user object
+		const user = await User.findBy('id', params.id);
+		var layoutType = '';
+
+		// Check type of user in order to render the correct layout
+		if (auth.user.role === 1) {
```

Comment:
You should check if the user is editing his own account

Reply:
isn't that already checked when the button is rendered, with the canEdit variable?

Reply:
that blocks the button from rendering but it doesn't plot someone from entering the page by just typing in the url

Reply:
That blocks the button from renedering but it doesnt block the user from entering by typing in the url